rust/azure_iot_operations_connector/src/filemount/azure_device_registry.rs

```diff
@@ -0,0 +1,1114 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Utilities for interacting with the file mount of Azure Device Registry.
+
+use std::collections::{HashMap, HashSet};
+use std::env;
+use std::fmt::{Display, Formatter};
+use std::path::{Path, PathBuf};
+use std::task::Context;
+use std::time::Duration;
+
+use notify::{
+    RecommendedWatcher,
+    event::{self, EventKind},
+};
+use notify_debouncer_full::{RecommendedCache, new_debouncer};
+use thiserror::Error;
+use tokio::sync::mpsc::{self, UnboundedReceiver, UnboundedSender};
+use tokio::sync::oneshot;
+
+/// Environment variable name for the mount path of the Azure Device Registry resources.
+const ADR_RESOURCES_NAME_MOUNT_PATH: &str = "ADR_RESOURCES_NAME_MOUNT_PATH";
+
+/// Represents an error that occurred while interacting with the file mount of Azure Device Registry.
+#[derive(Debug, Error)]
+#[error(transparent)]
+pub struct Error(#[from] ErrorKind);
+
+/// Represents the kinds of errors that may occur while interacting with the file mount of Azure Device Registry.
+#[derive(Error, Debug)]
+pub enum ErrorKind {
+    /// An error occurred when parsing the environment variable for the mount path.
+    #[error(transparent)]
+    EnvironmentVariableError(#[from] env::VarError),
+
+    /// An error occurred while accessing the file mount.
+    #[error(transparent)]
+    IoError(#[from] std::io::Error),
+
+    /// An error occurred while creating the file mount watcher.
+    #[error(transparent)]
+    WatcherError(#[from] notify::Error),
+
+    /// An error occurred while parsing the files in the file mount.
+    #[error("{0}")]
+    ParseError(String),
+}
+
+// ~~~~~~~~~~~~~~~~~ Observations ~~~~~~~~~~~~~~~~~~~~~
+
+/// Represents an observation for device endpoint creation events.
+#[allow(dead_code)]
```

Comment:
might not need this anymore since you have a sample (check for other instances of this as well)

Reply:
It was just here!